src/VisualStudio/Core/Def/Implementation/Interop/IComWrapperFactory.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.VisualStudio.Shell.Interop
+{
+    // TODO: Remove this definition of IComWrapperFactory and use the one from the VSSDK
+    // (Microsoft.VisualStudio.Shell.Interop.14.0.DesignTime) when it is available.
+    [ComImport, Guid("436b402a-a479-41a8-a093-9713ce3ad111"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown), TypeIdentifier]
+    public interface IComWrapperFactory
+    {
+        object CreateAggregatedObject(object managedObject);
```

Comment:
Curious, is this implemented in managed code in the shell (that ultimately calls into msenv.dll) or is it directly implemented in native? IIRC, once a CCW has already been created for an object it can't be aggregated with, which meant we had to be careful to use this mechanism very early in the creation of objects like this. Is this not a problem in this case (or is my memory failing in my old age)?